src/Microsoft.Health.ControlPlane.CosmosDb/Features/Storage/ControlPlaneDataStore.cs

```diff
@@ -118,6 +176,32 @@ private async Task<T> GetSystemDocumentByIdAsync<T>(string id, string partitionK
             }
         }
 
+        private async Task<IEnumerable<T>> GetSystemDocumentAllAsync<T>(CancellationToken cancellationToken)
+        {
+            var documentQuery = new SqlQuerySpec(
+                "SELECT * FROM root");
```

Comment:
This is going to get everything in the database. We should be able to make it a partitioned query and only return items in the `_roles` partition to get everything.